### ISSUE TITLE
Deprecate the codelab in favour of the Kubeflow GKE end-to-end guide

### ISSUE THREAD

**Issue by sarahmaddox** [googlecodelabs/kubeflow-introduction#7]
We now have an end-to-end GKE guide in the Kubeflow docs, which uses the code sample from this codelab.

Here's the new Kubeflow doc:
https://www.kubeflow.org/docs/guides/gke/gcp-e2e/

It'd be good to deprecate this codelab in favour of the Kubeflow docs and samples. I think the steps needed are:
1) Merge the code sample from this repo into the Kubeflow samples repo.
2) Deprecate the codelab tutorial:
https://codelabs.developers.google.com/codelabs/kubeflow-introduction/index.html#0

**Comment by sarahmaddox** [comment]
Thanks Dan, sounds good. One strategy would be to explain to customers that the codelabs are updated for specific events but not necessarily kept up to date outside the event timeframes, and to link to the corresponding tutorial for folks who're running the tutorial on their own.

**Comment by jlewi** [comment]
Can we have a single copy of the code in kubeflow/examples?

We can still have a codelab hosted at https://codelabs.developers.google.com that points to a fixed commit in kubeflow/examples.

We need to add tests: kubeflow/examples#460 and keep the code up to date as Kubeflow evolves. This is much harder if we have multiple copies of the code.

**Comment by DanSanche** [comment]
Yes, I believe the current plan is to migrate the code here into the examples repo as part of the codelab update to 0.4. There's a meeting set up to go through the details tomorrow. I see you're on the calendar invite, but I'll update here if there are any new decisions made

**Comment by jlewi** [comment]
Here's some additional thoughts on how to reconcile the code

* I think we should use https://github.com/kubeflow/examples/blob/master/mnist/model.py and not https://github.com/googlecodelabs/kubeflow-introduction/blob/master/tensorflow-model/MNIST.py

  * The former is using tf.estimator
  * The latter is not
  * I think tf.estimator is the preferred pattern; MNIST.py is doing a loop in feeding in the examples
      * This is less performant because input can't be pulled in batches into TF.

* It looks like kubeflow/examples is missing the web app front end so we should copy that over.

**Comment by happyhuman** [comment]
I was actually planning to use tensorflow keras to implement a much simpler and shorter example for mnist (considering that keras will be the recommendation of tensorflow 2.0 anyway). I think the emphasis of this kubeflow example will be to show how to use kubeflow, and not so much on how to train a model anyway. But I am open to other ideas as well.

**Comment by jlewi** [comment]
I think if you want to change the model to use Keras that's fine. Swapping out the model once we have tests should be fine because the tests will ensure the updates don't break it.

**Comment by jlewi** [comment]
Serving is now updated in kubeflow/examples/mnist.

I'm working on moving over the web-ui and then I think the copy in kubeflow/examples will be up to date and contain everything we want to preverse from googlecodelabs. We can then proceed with moving to Keras or other improvements as desired.

One difference is; codelab is walking through building the ksonnet app
https://codelabs.developers.google.com/codelabs/kubeflow-introduction/index.html#4

Whereas in kubeflow/examples/mnist I'm checking in a ksonnet app and user just needs to set parameters needed to customize it for their particular install
